client-device-connection-infinispan/src/main/java/org/eclipse/hono/deviceconnection/infinispan/client/HotrodBasedDeviceConnectionInfoCache.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+
+package org.eclipse.hono.deviceconnection.infinispan.client;
+
+import java.net.HttpURLConnection;
+import java.util.Objects;
+
+import org.eclipse.hono.client.ClientErrorException;
+import org.eclipse.hono.client.ServerErrorException;
+import org.eclipse.hono.util.DeviceConnectionConstants;
+import org.eclipse.hono.util.HonoProtonHelper;
+import org.infinispan.commons.api.BasicCache;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.opentracing.SpanContext;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+
+
+/**
+ * A client for accessing device connection information in a data grid using the
+ * Hotrod protocol.
+ *
+ */
+public final class HotrodBasedDeviceConnectionInfoCache implements DeviceConnectionInfoCache {
+
+    private static final Logger LOG = LoggerFactory.getLogger(HotrodBasedDeviceConnectionInfoCache.class);
+
+    final BasicCache<String, String> cache;
+
+    /**
+     * Creates a client for accessing device connection information.
+     * 
+     * @param cache The remote cache that contains the data.
+     */
+    public HotrodBasedDeviceConnectionInfoCache(final BasicCache<String, String> cache) {
+        this.cache = Objects.requireNonNull(cache);
```

Comment:
IMHO this constructor throws `NullPointerException`, if the _cache_ is _null_ and this info could be added in the _javadoc_.